server/downloader.cpp

```diff
@@ -6,28 +6,30 @@
 #include <spdlog/spdlog.h>
 #include "downloader.h"
 
-downloader::downloader(std::string path, std::string server)
-        : valid_(false),
-          path_(std::move(path)),
-          server_(std::move(server)) {
-
-    spdlog::info("create downloader, path: {}, server: {}", path_, server_);
-    if (server_.empty() || path_.empty()) {
-        spdlog::error("invalid downloader, path: {}, server: {}", path_, server_);
-        return;
-    }
 
-    server_split_ = split_server_name();
-    if (server_split_.first.empty()) {
-        spdlog::error("split server name failed, server: {}", server_);
+std::string downloader::pdb_path_="";
+std::pair<std::string,std::string> downloader::server_split_= std::make_pair("","");
+std::string downloader::msdl_server_="";
+
+void downloader::set_default_pdb_path(std::string dir_path) {
+    downloader::pdb_path_=dir_path;
+    spdlog::info("set_default_pdb_path: {}", dir_path);
+}
+
+void downloader::set_default_pdb_server(std::string msdl_server) {
+    spdlog::info("set_default_pdb_server: {}", msdl_server);
+    std::regex regex(R"(^((?:(?:http|https):\/\/)?[^\/]+)(\/.*)$)");
+    std::smatch match;
+    if (!std::regex_match(msdl_server, match, regex)) {
+        spdlog::info("invaild server url: {}",msdl_server);
         return;
     }
-
-    valid_ = true;
+    downloader::server_split_= std::make_pair(match[1].str(), match[2].str());
+    downloader::msdl_server_=msdl_server;
 }
 
-bool downloader::valid() const {
-    return valid_;
+downloader::downloader(){
```

Comment:
please remove the empty constructor. if we don't write a constructor, C++ creates an implicit constructor with no parameters